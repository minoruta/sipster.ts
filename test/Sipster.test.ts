import { Sipster,
AudioMediaPlayer,
AudioMediaRecorder,
CallInfo,
Transport,
Account
} from '../src/Sipster';

let sipster: Sipster;
let transport: Transport;
let account: Account;
let player: AudioMediaPlayer;
let recorder: AudioMediaRecorder;
const events: {
    registering: number;
    unregistering: number;
    registered: number;
    unregistered: number;
    state: number,
    call: number
} = {
    registering: 0,
    unregistering: 0,
    registered: 0,
    unregistered: 0,
    state: 0,
    call: 0
};

function delay(sec: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, sec * 1000));
}

describe('sipster.ts', () => {
    test ('check static properties', () => {
        expect(Sipster.version).toMatchSnapshot();
    });
    test ('create a Sipster', () => {
        const configs = { logConfig: { level: 6, consoleLevel: 6 }};
        sipster = Sipster.instance(configs);
        expect(Sipster.instance()).toBe(sipster);
        expect(sipster.config).toMatchSnapshot();
        expect(sipster.mediaActivePorts).toMatchSnapshot();
        expect(sipster.mediaMaxPorts).toMatchSnapshot();
        expect(sipster.state).toMatchSnapshot();
    });
    test ('set up a transport to listen for incoming connections, defaults to UDP', () => {
        transport = new sipster.Transport({ port: 5060 });
        const info = transport.getInfo();
        info.info = undefined;
        info.localName = undefined;
        expect(info).toMatchSnapshot();
        expect(transport.enabled).toMatchSnapshot();
    });
    test ('set up a SIP account, we need at least one', () => {
        // this sets up an account for calls coming from 192.168.100.10
        account = new sipster.Account({ idUri: 'sip:192.168.100.10' });
        expect(account.getInfo()).toMatchSnapshot();
        expect(account.valid).toMatchSnapshot();
        expect(account.default).toMatchSnapshot();
    });
    test ('set up listener', () => {
        account.on('registering', () => events.registering++);
        account.on('unregistering', () => events.unregistering++);
        account.on('registered', () => events.registered++);
        account.on('unregistered', () => events.unregistered++);
        account.on('state', (active: boolean, statusCode: number) => events.state++);
        account.on('call', (info: boolean, call: CallInfo) => events.call++);
    });
    test ('make an audio player', () => {
        player = sipster.createPlayer(`${__dirname}/waves/sound.wav`);
    });
    test ('make an audio recorder', () => {
        recorder = sipster.createRecorder(`${__dirname}/waves/call.wav`);
    });
    test ('finalize the pjsip initialization phase', () => {
        sipster.start();
    });
    test ('wait for any events', async () => {
        await delay(3);
        expect(sipster.state).toMatchSnapshot();
        expect(events).toMatchSnapshot();
    });
});
