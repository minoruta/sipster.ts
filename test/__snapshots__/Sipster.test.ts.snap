// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sipster.ts check static properties 1`] = `
Object {
  "full": "2.7.1",
  "major": 2,
  "minor": 7,
  "numeric": 34013440,
  "rev": 1,
  "suffix": "",
}
`;

exports[`sipster.ts create a Sipster 1`] = `
Object {
  "EpConfig": Object {
    "LogConfig": Object {
      "consoleLevel": 3,
      "decor": 25328,
      "fileFlags": 0,
      "filename": "",
      "level": 3,
      "msgLogging": 1,
    },
    "MediaConfig": Object {
      "audioFramePtime": 20,
      "channelCount": 1,
      "clockRate": 16000,
      "ecOptions": 0,
      "ecTailLen": 200,
      "hasIoqueue": true,
      "ilbcMode": 30,
      "jbInit": -1,
      "jbMax": -1,
      "jbMaxPre": -1,
      "jbMinPre": -1,
      "maxMediaPorts": 254,
      "noVad": false,
      "ptime": 0,
      "quality": 8,
      "rxDropPct": 0,
      "sndAutoCloseTime": 1,
      "sndClockRate": 0,
      "sndPlayLatency": 140,
      "sndRecLatency": 100,
      "threadCnt": 1,
      "txDropPct": 0,
      "vidPreviewEnableNative": true,
    },
    "UaConfig": Object {
      "mainThreadOnly": false,
      "maxCalls": 4,
      "mwiUnsolicitedEnabled": true,
      "nameserver": Array [],
      "natTypeInSdp": 1,
      "stunIgnoreFailure": true,
      "stunServer": Array [],
      "stunTryIpv6": false,
      "threadCnt": 1,
      "userAgent": "",
    },
  },
}
`;

exports[`sipster.ts create a Sipster 2`] = `1`;

exports[`sipster.ts create a Sipster 3`] = `254`;

exports[`sipster.ts create a Sipster 4`] = `"init"`;

exports[`sipster.ts set up a SIP account, we need at least one 1`] = `
Object {
  "regExpiresSec": -1,
  "regIsActive": false,
  "regIsConfigured": false,
  "uri": "sip:192.168.100.10",
}
`;

exports[`sipster.ts set up a SIP account, we need at least one 2`] = `true`;

exports[`sipster.ts set up a SIP account, we need at least one 3`] = `true`;

exports[`sipster.ts set up a transport to listen for incoming connections, defaults to UDP 1`] = `
Object {
  "flags": 4,
  "info": undefined,
  "localAddress": "0.0.0.0:5060",
  "localName": undefined,
  "type": "UDP",
  "usageCount": 1,
}
`;

exports[`sipster.ts set up a transport to listen for incoming connections, defaults to UDP 2`] = `true`;

exports[`sipster.ts wait for any events 1`] = `"running"`;

exports[`sipster.ts wait for any events 2`] = `
Object {
  "call": 0,
  "registered": 0,
  "registering": 0,
  "state": 0,
  "unregistered": 0,
  "unregistering": 0,
}
`;
